python/tvm/relay/ir_pass.py

```diff
@@ -491,9 +493,18 @@ def gradient(expr, mod=None):
     mod : Optional[tvm.relay.Module]
         The global module.
 
+    mode : Optional[String]
+        The mode of the automatic differentiation algorithm.
+        'first_order' only work on first order code, but will not produce reference nor closure.
+        'higher_order' work on all code using reference and closure.
     Returns
     -------
     ret : tvm.relay.Expr
-        A function that calculate the original result paired with gradient.
+        The transformed expression.
     """
-    return _ir_pass.first_order_gradient(expr, mod)
+    if mode == 'first_order':
+        return _ir_pass.first_order_gradient(expr, mod)
+    elif mode == 'higher_order':
+        return _ir_pass.gradient(expr, mod)
```

Comment:
Quick question: why not just expose higher-order gradient?

Reply:
first order work faster by not using ref. we might optimize the ref away though.